### ISSUE TITLE
Is it wrong to do softmax before sparse_softmax_cross_entropy_with_logits in MDPRank.py?

### ISSUE THREAD

**Issue by kelvict** [wzeng316/MDPRank#1]
Is it wrong to do softmax before sparse_softmax_cross_entropy_with_logits in MDPRank.py?

        prob = tf.nn.softmax(scores)
        init = tf.global_variables_initializer()
        cross_entropy = tf.nn.sparse_softmax_cross_entropy_with_logits(logits=prob, labels=position)
